### DISCUSSION TITLE
Label Get Entities don't work at all?

### DISCUSSION THREAD

**Discussion by degracode** [zachowj/node-red-contrib-home-assistant-websocket#1509] [Q&A] [answered]
I've been trying out labels + Get Entities for the first time.
For any label I test, it just returns 'No entities', whether by filtering by name or label_id.

I've also found that the Action node doesn't work in the same way. Performing the same label-based service call via a regular HA automation works fine.

Is there another step, or something else I need to upgrade to get the new label features on these nodes to work?

I've gone for a Discussion rather than an Issue, because I assume the new feature _does_ work (otherwise it wouldn't have been pushed!) and that there's something else going on...

**Comment by zachowj** [comment]
I just tested the get-entities and action nodes. Both are doing what I would expect them to do.

![firefox_H45HlY8QvR](https://github.com/user-attachments/assets/6093709a-3404-41ec-9466-5661287bea08)

**Reply by degracode** [reply]
Thanks for the reassurance - I'll give it another bash.

**Reply by zachowj** [reply]
The action node should be easy as it will only show labels that are valid for the action.

**Reply by degracode** [reply]
I've figured it out - it's the difference between setting a label on an entity vs a device.
I added the label to my zigbee light devices, not the light entities within them. Once I added a label to the entity, it showed up.

**Reply by zachowj** [reply, marked as answer]
Thank you for pointing that out. I realize now that the action node logic for listing labels is flawed. It should display labels when they are assigned to valid devices, areas, or floors, not just based on the entity ID.

As for the get-entities node, it’s working as expected since it only targets entities.

**Comment by LeiDuang** [comment]
I have the same problem, I can't get the entity through label, please tell me how to solve this kind of problem in the end

**Reply by zachowj** [reply]
@LeiDuang Can you explain what you're trying to do and post an example of what you have tried?

**Reply by LeiDuang** [reply]
I'm using the get entities component to get all the entities under the specified label, but when I try to run it, no entities are found, so I don't know if there's a problem with the functionality or with my setup, what I can rule out right now is that the nodered connection to HA is working fine, and that the get entities component is following the entity_id is also working fine, just getting it by label is failing, and I'm sure the nodered plugin is up to date.

![image](https://github.com/user-attachments/assets/a6480efa-cbca-4e9e-ab39-d9b00565d2af)

**Reply by zachowj** [reply]
Have you tried using the `label_id`? You can easily find it by selecting the label in an action node's target selection, and then using the copy button to grab the label's ID.

**Reply by LeiDuang** [reply]
I tried to get the tag name “deng_ju” in the action component and I'm sure it contains a lot of entities, but the result is still empty

![image](https://github.com/user-attachments/assets/db445a63-8492-4b48-8917-b5f18e538fc8)

**Reply by zachowj** [reply]
What version are you running?

<https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/diagnostics.html>

**Reply by LeiDuang** [reply]
Version: 0.73.0

Home Assistant version: 2024.9.3
Companion version: 4.1.1
node-red-contrib-home-assistant-websocket：0.73.0

Node-RED version: 4.0.2
Docker: yes
Add-on: no

Node.js version: v20.17.0 x64 linux
OS: Linux 4.4.302+ x64
